packages/main/src/plugin/container-registry.ts

```diff
@@ -2235,8 +2241,27 @@ export class ContainerProviderRegistry {
 
       const imageObject = provider.api.getImage(id);
       if (imageObject) {
-        const imageStream = await imageObject.get();
-        return pipeline(imageStream, fs.createWriteStream(filename));
+        // make the download of image cancellable
+        const getImageObjectPromise = imageObject.get();
+        const cancelPromise = new Promise<NodeJS.ReadableStream>((_, reject) => {
+          token?.onCancellationRequested(() => {
+            reject(new Error('saveImage operation canceled'));
+          });
+        });
+        const imageStream = await Promise.race<NodeJS.ReadableStream>([getImageObjectPromise, cancelPromise]);
```

Comment:
This does not cancel the `image.get` call, it give the illusion but does not cancel the request.

As visible in the dockerode implementation of the image get, they do not support cancelling this request.

https://github.com/apocas/dockerode/blob/49bd8c4c231efe9bb432b28d51dbae3e50ddd010/lib/image.js#L125-L135

You can compare to other method which support to have an `AbortSignal`

https://github.com/apocas/dockerode/blob/49bd8c4c231efe9bb432b28d51dbae3e50ddd010/lib/image.js#L160-L172

If this call is not cancellable, it might be because it is a very small operation, which execute almost instantly, therefore would not need this racing logic.

If the call is something which can be heavy, it would be better to open a pull request on their side to add the signal option support.

Then on https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/dockerode adding the type for the signal argument.

Reply:
Yes, I agree that the call to `image.get` itself is not canceled. 

But my intention here is to at least 'detach' from this call, so the function can continue, while the call to `image.get` continues on its side. This call having side effects only in memory, I don't feel that there is any incidence (expect of some optimization).

The most important part for me is the second part, where the call to `pipeline` is effectively aborted, which is important because we want to stop the writing on disk before to delete the file.